caput/memh5.py

```diff
@@ -2152,40 +2187,89 @@ def _map_json(value):
 
 
 def deep_group_copy(
-    g1, g2, convert_dataset_strings=False, convert_attribute_strings=False
+    g1,
+    g2,
+    selections=None,
+    convert_dataset_strings=False,
+    convert_attribute_strings=False,
 ):
-    """Copy full data tree from one group to another."""
+    """
+    Copy full data tree from one group to another.
+
+    Copies from g1 to g2. An axis downselection can be specified by supplying the
+    parameter 'selections'. If for example 'g1' is
+
+    ```
+    foo:
+       bar: [1, 2, 3]
+    data: [0.1, 0.2, 0.3]
+    ```
+    , calling
+    ```
+    deep_group_copy(g1, g2, {"foo": {"bar": slice(2)}})
+    ```
+    would select the first two indexes in g1["foo"]["bar"], resulting in the
+    following 'g2':
+
+    ```
+    foo:
+       bar: [0,1]
+    data: [0.1, 0.3]
+    ```
+
+    Parameters
+    ----------
+    g1 : h5py.Group
+        Deep copy from this group.
+    g2 : h5py.Group
+        Deep copy to this group.
+    selections : dict
+        If this is not None, it should have a subset of the same hierarchical structure
+        as g1, but ultimately describe axis selections for group entries as valid
+        numpy indexes.
+    convert_strings : bool, optional
+        Convert string attributes (or lists/arrays of them) to ensure that they are
+        unicode.
+    """
+
+    if selections is None:
```

Comment:
Can't you remove this along with the branch commented on below.

Reply:
You can't pass `None` as a numpy index.

Reply:
I could make the default argument `slice(None)` but I want it to be pretty.